### PR TITLE
58721_login_logout

### PR DESCRIPTION
**Pull request by ahammad-reyad**
Login Logout flow implemented

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def current_user
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by(id: user_id)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    elsif user_id = cookies.signed[:user_id]
+      user = User.find_by(id: user_id)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        @current_user = user
+      end
+    end
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def forget(user)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/models/user.rb [hunk 4]
```diff
@@ -17,4 +18,29 @@ class User < ActiveRecord::Base
...
+      BCrypt::Password.create string, cost: cost
+  end
+
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attributes remember_digest: User.digest(remember_token)
```
Comment (dieunb) [inline app/models/user.rb]: I want to you research and understand about difference between `update_attributes` and `update_attribute`

Comment (ahammad-reyad) [inline app/models/user.rb]: update_attribute is used to update single attribute of object without invoking model based validation and on the other hand update_attributes update multiple attribute of single object and also pass model based validation. Thank you for your guidance.

File: app/helpers/sessions_helper.rb (hunk 1 above)
Comment (ahammad-reyad) [inline app/helpers/sessions_helper.rb]: removed

File: app/helpers/sessions_helper.rb (hunk 2 above)
Comment (ahammad-reyad) [inline app/helpers/sessions_helper.rb]: removed

File: app/helpers/sessions_helper.rb (hunk 3 above)
Comment (ahammad-reyad) [inline app/helpers/sessions_helper.rb]: removed

File: app/models/user.rb [hunk 5]
```diff
@@ -17,4 +18,29 @@ class User < ActiveRecord::Base
...
+      BCrypt::Password.create string, cost: cost
+  end
+
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+     update_attributes remember_digest: User.digest(remember_token)
```
Comment (MuhammadTamzid) [inline app/models/user.rb]: indent

**Pull request merged**